client/components/progress/index.tsx

```diff
@@ -1,41 +1,14 @@
-import React, { useEffect, useState } from 'react';
-import { useSelector } from 'react-redux';
 import { Progress } from 'native-base';
-import useTheme from '~/hooks/useTheme';
+import useProgressBarComponent from '~/hooks/progress';
 
 const ProgressBarComponent = () => {
-  const { currentTheme } = useTheme();
-  const reduxTargetValue = useSelector((state) => state.progress.targetValue);
-  const [localCurrentValue, setLocalCurrentValue] = useState(0);
-  const [operationId, setOperationId] = useState(0);
 
-  useEffect(() => {
-    // Generate a new operation ID
-    const newOperationId = Math.random();
-    setOperationId(newOperationId);
-
-    // Reset local current value to start new progress
-    setLocalCurrentValue(0);
-
-    function incrementProgress() {
-      if (
-        localCurrentValue < reduxTargetValue &&
-        operationId === newOperationId
-      ) {
-        setLocalCurrentValue((prevValue) => prevValue + 1);
-      }
-    }
-
-    // Start the increment
-    incrementProgress();
-  }, [reduxTargetValue]);
-
-  useEffect(() => {
-    if (localCurrentValue < reduxTargetValue && operationId !== 0) {
-      setLocalCurrentValue((prevValue) => prevValue + 1);
-    }
-  }, [localCurrentValue, reduxTargetValue, operationId]);
+  const {
+    localCurrentValue,
+    currentTheme
```

Comment:
the useTheme logic can actually stay in the component, only because we will be refactoring these component to use tamagui theme here soon. No worries, i did not explain that detail well. nice job :)